### DISCUSSION TITLE
Ignore 'pending' telegrams?

### DISCUSSION THREAD

**Discussion by proddy** [emsesp/EMS-ESP32#1344] [Ideas]
@MichaelDvP 

Shall we, after some time, add the 'Pending' telegrams to the 'Ignored' list?

Otherwise, we'll be constantly 'fetching' these telegrams forever, correct?

```
These EMS devices are currently active:

boiler: Nefit GBx72/Trendline/Cerapur/Greenstar Si (DeviceID:0x08, ProductID:123, Version:06.01)
 Received telegram type IDs: 0x10 0x11 0x15 0x1C 0x18 0x19 0x34 0x2A 0x04
 Fetched telegram type IDs: 0x14 0x16 0x33 0x26
 Pending telegram type IDs: 0xBF 0xC2 0x1A 0x35 0xD1 0xE3 0xE4 0xE5 0xE6 0xE9 0xEA
 Ignored telegram type IDs:

thermostat: RC20/Moduline 300 (DeviceID:0x17, ProductID:77, Version:03.03)
 Received telegram type IDs: 0xA3 0x06 0xA2
 Fetched telegram type IDs: 0x91 0xA8 0x90 0x8F
 Pending telegram type IDs: 0x12 0x13 0xAF
 Ignored telegram type IDs: 0x1A 0x35
```

**Comment by MichaelDvP** [comment]
> Otherwise, we'll be constantly 'fetching' these telegrams forever, correct?

We are only the fetching the telegrams in the "Fetched" list, "Pending" is for registered telegrams not yet received. 

Especialy for boilers we can use them to use/add deviceTypes and register less telegram-types. 
We have EMS_DEVICE_FLAG_EMS, but don't know which boilers are in that category (actually only mine). So most boilers have EMS_DEVICE_FLAG_NONE and register all telegrams. 

<s>But you're right, we do not check if the fetched telegrams give a valid reply. For empty messages with offset 0 we can remove them from fetch.</s>

Edit: we already remove from fetch, see here:
https://github.com/emsesp/EMS-ESP32/blob/f9a176e09c55bb9cc23f3f41a05aa45300c82a19/src/emsdevice.cpp#L1787C20-L1787

**Reply by proddy** [reply]
I was confused with my setup above with an RC20 thermostat, and this line:
`Fetched telegram type IDs: 0x91 0xA8 0x90 0x8F`

but when I look at the code in https://github.com/emsesp/EMS-ESP32/blob/f9a176e09c55bb9cc23f3f41a05aa45300c82a19/src/devices/thermostat.cpp#L85 

the fetch flag is set to off. How is this then fetched?

The reason I was asking is that I was missing some thermostat entities from the 0x90 telegram like day and night temperatures. They only appeared when I rebooted EMS-ESP.
 
(By the way, it's years since I wrote the core of this code and can't remember most of it. It seemed I was a lot smarter back then.)

**Reply by MichaelDvP** [reply]
The fetch flag for thermostat is set on first monitor message received and heatingcircuit is created: 
https://github.com/emsesp/EMS-ESP32/blob/f9a176e09c55bb9cc23f3f41a05aa45300c82a19/src/devices/thermostat.cpp#L388-L389
The `Fetched telegram type IDs` scans the actual status of the fetch flag.
